src/wire_msg.rs

```diff
@@ -41,7 +42,16 @@ impl WireMsg {
 
         let msg_header = MsgHeader::from_bytes(header_bytes);
         log::debug!("reading data of length: {}", msg_header.data_len());
-        let mut data: Vec<u8> = vec![0; msg_header.data_len()];
+        // https://github.com/rust-lang/rust/issues/70460 for work on a cleaner alternative:
+        #[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
+        {
+            compile_error!("You need an architecture capable of addressing 32-bit pointers");
+        }
+        let data_length = match usize::try_from(msg_header.data_len()) {
+            Ok(s) => s,
+            Err(_) => return Err(Error::MaxLengthExceeded(usize::MAX)), // this should never happen due to preceding compilation error
```

Comment:
Just realized that this new use (where it gives the maximum allowed length) as part of `MaxLengthExceeded` conflicts with the existing use (where it gives the specified length).

Which do we prefer? If the latter, this error needs to change types to `u32` or `u64` or something to be portable, but I believe that would be a breaking change?
I can also create a new dummy error type called "unreachable" or something and indicate that the user should file a bug report.

Reply:
I think the only way to not change behavior here is to make a new error type.